include/seqan3/argument_parser/validators.hpp

```diff
@@ -412,6 +413,20 @@ class input_file_validator : public file_validator_base
     input_file_validator & operator=(input_file_validator &&) = default;      //!< Defaulted.
     virtual ~input_file_validator() = default;                                //!< Virtual destructor.
 
+    /*!\brief Constructs from a given file.
+     * \tparam    file_t The type of the file; must have a member type called `valid_formats` containing the formats.
+     * \param[in] file   The file to get the valid extensions for.
+     *
+     * \details
+     *
+     * Uses the extensions from the formats of the file to initialise the valid extensions.
+     */
```

Comment:
yes, I'll do

Reply:
well, if you consider it they are not usable at all. You would need to create a file but obviously you don't have a path yet since nothing was parsed yet. So the only way seems to to provide the file information as template argument.